core/frontend/src/tools/ElementSetTool.ts

```diff
@@ -307,25 +204,705 @@ export class ElementAgenda {
+        ctx.fillStyle = bestContrastIsBlack ? "rgba(0,0,0,.06)" : "rgba(255,255,255,.06)";
+        ctx.fillRect(0, 0, offset.x, offset.y);
+      }
+    };
+    context.addCanvasDecoration({ position, drawDecoration });
+  }
+
+  public decorate(context: DecorateContext): void { this.selectByPointsDecorate(context); }
+
+  public async onMouseMotion(ev: BeButtonEvent): Promise<void> {
+    if (undefined !== ev.viewport && this.isSelectByPoints)
+      ev.viewport.invalidateDecorations();
+  }
+
+  public async onMouseStartDrag(ev: BeButtonEvent): Promise<EventHandled> {
+    if (await this.selectByPointsStart(ev))
+      return EventHandled.Yes;
+    return super.onMouseStartDrag(ev);
+  }
+
+  public async onMouseEndDrag(ev: BeButtonEvent): Promise<EventHandled> {
+    if (await this.selectByPointsEnd(ev))
+      return EventHandled.Yes;
+    return super.onMouseEndDrag(ev);
+  }
+
+  public async onModifierKeyTransition(_wentDown: boolean, modifier: BeModifierKeys, _event: KeyboardEvent): Promise<EventHandled> {
+    if (this.isSelectionSetModify)
+      return EventHandled.No;
+
+    if (this.isSelectByPoints)
+      return (BeModifierKeys.Shift === modifier ? EventHandled.Yes : EventHandled.No);
+
+    if (BeModifierKeys.Control !== modifier || undefined !== this.anchorPoint || !this.controlKeyContinuesSelection)
+      return EventHandled.No;
+
+    if (this.currentElementCount < this.requiredElementCount && !this.wantAccuSnap)
+      return EventHandled.No; // Can only early return if AccuSnap doesn't need to be disabled for ctrl selection...
+
+    this.setupAndPromptForNextAction(); // Enable/disable auto-locate, AccuSnap, update prompts...
+    return EventHandled.Yes;
+  }
+
+  /** Advance to next prelocated hit using auto-locate or change last accepted hit to next hit on a reset button event.
+   * @returns EventHandled.Yes if onReinitalize was called to restart or exit tool.
+   */
+  protected async chooseNextHit(ev: BeButtonEvent): Promise<EventHandled> {
+    if (this.isSelectionSetModify) {
+      this.onReinitialize();
+      return EventHandled.Yes;
+    }
+
+    if (0 !== this.currentElementCount) {
+      let autoLocateChooseNext = false;
+
+      if (this.wantAdditionalElements) {
+        const lastHit = IModelApp.locateManager.currHit;
+        const autoHit = IModelApp.accuSnap.currHit;
+
+        // Choose next using auto-locate or normal locate?
+        if (undefined !== autoHit && (undefined === lastHit || !autoHit.isSameHit(lastHit)))
+          autoLocateChooseNext = true;
+      }
+
+      if (!autoLocateChooseNext) {
+        await this.doLocate(ev, false);
+
+        if (this.agenda.isEmpty) {
+          this.onReinitialize();
+          return EventHandled.Yes;
+        }
+
+        this.setupAndPromptForNextAction();
+        return EventHandled.No;
+      }
+    }
+
+    IModelApp.accuSnap.resetButton(); // eslint-disable-line @typescript-eslint/no-floating-promises
```

Comment:
Is there a reason not to `await` this?

Reply:
Just being consistent with SelectTool, there's no need to await...